addons/mail/static/src/models/link_preview.js

```diff
@@ -0,0 +1,93 @@
+/** @odoo-module **/
+
+import { registerModel } from '@mail/model/model_core';
+import { attr, many, one } from '@mail/model/model_field';
+
+registerModel({
+    name: 'LinkPreview',
+    identifyingFields: ['id'],
+    recordMethods: {
+        async remove() {
+            if (this.composer) {
+                this.composer.linkPreviewsRemovedUrls.push(this.url);
+                this.composer.update({ linkPreviewsRemovedUrls: this.composer.linkPreviewsRemovedUrls });
+            }
+            await this.messaging.rpc({
+                route: '/mail/link_preview/delete',
+                params: { link_preview_id: this.id },
+            }, { shadow: true });
+        },
+        /**
+         * @private
+         * @returns {boolean}
+         */
+        _computeIsGif() {
+            return this.mimetype === 'image/gif';
+        },
+        /**
+         * @private
+         * @returns {boolean}
+         */
+        _computeIsImage() {
+            return this.type === 'image' && !this.isGif;
+        },
+        /**
+         * @private
+         * @returns {boolean}
+         */
+        _computeIsVideo() {
+            const type = this.type.split('.');
+            return type.includes('video') && !this.isGif;
+        },
+        /**
+         * @private
+         * @returns {boolean}
+         */
+        _computeIsCard() {
+            return !this.isVideo && !this.isGif && !this.isImage;
+        }
+    },
+    fields: {
+        id: attr({
+            required: true,
+            readonly: true,
+        }),
+        composer: one('Composer', {
+            inverse: 'linkPreviews',
+        }),
+        linkPreviewCardView: many('LinkPreviewCardView', {
+            inverse: 'linkPreview',
+            isCausal: true,
+        }),
+        linkPreviewGifView: many('LinkPreviewGifView', {
+            inverse: 'linkPreview',
+            isCausal: true,
+        }),
+        linkPreviewImageView: many('LinkPreviewImageView', {
+            inverse: 'linkPreview',
+            isCausal: true,
+        }),
+        linkPreviewVideoView: many('LinkPreviewVideoView', {
+            inverse: 'linkPreview',
+            isCausal: true,
+        }),
+        isImage: attr({
+            compute: '_computeIsImage',
+        }),
+        isGif: attr({
+            compute: '_computeIsGif',
+        }),
+        isVideo: attr({
+            compute: '_computeIsVideo',
+        }),
+        isCard: attr({
+            compute: '_computeIsCard',
+        }),
+        url: attr(),
+        title: attr(),
+        type: attr(),
+        image_url: attr(),
+        description: attr(),
+        mimetype: attr(),
```

Comment:
alphabetical etc but they will probably be all renamed or removed according to other suggestions